### PR TITLE
Add 'dry' option for the tests, plans and stories create method

### PR DESCRIPTION
**Pull request by adka1408**
Resolves the issue #366.

### CODE REVIEW

File: tmt/base.py [hunk 1]
```diff
@@ -286,35 +286,46 @@ def overview(tree):
...
-            name='test script', force=force, mode=0o755)
+
+        if Test._opt('dry'):
+            print(f'Directory {directory_path} was not created.')
+            print(f'Test metadata {metadata_path} was not created.')
+            print(f'Test script {script_path} was not created.')
+            print('No changes in workdir were executed.')
+        else:
+            # Create directory
+            tmt.utils.create_directory(directory_path, 'test directory')
```
Comment (psss) [inline tmt/base.py:309]: Instead of handling the `dry` mode three times separately in `Story`, `Plan` and `Test` it would be better to add support for dry mode to `tmt.utils.create_directory()` and `tmt.utils.create_file()` and then call them from here like this:

    tmt.utils.create_directory(directory_path, 'test directory', dry=Test._opt('dry))

This would save us a lot of unnecessary code duplication.

**Review by psss — CHANGES_REQUESTED** [review]

File: tmt/utils.py [hunk 2]
```diff
@@ -852,34 +852,42 @@ def format(
...
     except OSError as error:
         raise FileError("Failed to create {} '{}' ({})".format(
             name, path, error))
 
 
-def create_file(path, content, name, force=False, mode=0o664, quiet=False):
+def create_file(path, content, name, dry=False, force=False, mode=0o664, quiet=False):
     """ Create a new file, handle errors """
     say = log.debug if quiet else echo
     action = 'created'
```
Comment (psss) [inline tmt/utils.py:875]:
```suggestion
    action = 'would be created' if dry else 'created'
```
What about something like this? If we adjust the wording here (and similarly for `overwritten`) the rest of the code would be a little bit more clear.

File: tmt/utils.py [hunk 3]
```diff
@@ -852,34 +852,42 @@ def format(
...
+        else:
+            os.makedirs(path, exist_ok=True)
+            say("Directory '{}' created.".format(path))
     except OSError as error:
         raise FileError("Failed to create {} '{}' ({})".format(
             name, path, error))
 
 
-def create_file(path, content, name, force=False, mode=0o664, quiet=False):
+def create_file(path, content, name, dry=False, force=False, mode=0o664, quiet=False):
```
Comment (psss) [inline tmt/utils.py:872]: Please, break the long line (we use the pep8 limit of 79 chars).

File: tmt/utils.py [hunk 4]
```diff
@@ -852,34 +852,42 @@ def format(
...
+def create_directory(path, name, dry=False, quiet=False):
     """ Create a new directory, handle errors """
     say = log.debug if quiet else echo
     if os.path.isdir(path):
         say("Directory '{}' already exists.".format(path))
         return
     try:
-        os.makedirs(path, exist_ok=True)
-        say("Directory '{}' created.".format(path))
+        if dry:
```
Comment (psss) [inline tmt/utils.py:862]: In order to prevent unnecessary nesting let's move the `if dry` part above `try` and call `return` at the end of the block.

File: tmt/base.py [hunk 5]
```diff
@@ -656,6 +660,8 @@ def _match(
     @staticmethod
     def create(name, template, tree, force=False):
         """ Create a new story """
+        dry = True if Story._opt('dry') else False
```
Comment (psss) [inline tmt/base.py:663]: `Story._opt('dry')` already returns a `bool` so there's no need to check/convert it here.

```suggestion
        dry = Story._opt('dry')
```

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for improving the patch. Looks fine and works as expected. A couple of minor improvements suggested. Could you please also extend the `/tests/test/create` test to cover the `--dry` scenarios as well? Thanks!

File: tmt/base.py [hunk 6]
```diff
@@ -286,20 +286,23 @@ def overview(tree):
...
         """ Create a new test """
+        dry = Test._opt('dry')
+
         # Create directory
         if name == '.':
             directory_path = os.getcwd()
         else:
             directory_path = os.path.join(tree.root, name.lstrip('/'))
-            tmt.utils.create_directory(directory_path, 'test directory')
+        tmt.utils.create_directory(directory_path, 'test directory', dry=dry)
```
Comment (psss) [inline tmt/base.py:296]: Why is this line moved out of the `else` clause? When creating a new test in the current directory we do not need to create it. It already exists.

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for the changes. Looks very good now. Just one comment.
Could you also extend the [/tests/test/create](https://github.com/psss/tmt/tree/master/tests/test/create) so that it covers the dry functionality?

File: tmt/base.py (hunk 6 above)
Comment (adka1408) [reply tmt/base.py:296]: I'm sorry, I did it unintentionally and didn't notice that. I will fix!

**Review by psss — APPROVED** [review]
Thanks for the changes. Looks very good now, added just a couple of minor adjustments in afbc4fa and rebased on the latest master.
